# Análisis exploratorio PIB

Registro anual desde 1960 de PIB (GDP) para todos los países del mundo.
> PIB: es el monto total en USD de los bienes y servcios generados por un país en un año determinado

In [24]:
import pandas as pd

In [25]:
pib = pd.read_csv('datos/API_NY.GDP.MKTP.CD_DS2_en_csv_v2_397725.csv',
                skiprows=4)
pib

,Country Name,Country Code,Indicator Name,Indicator Code,1960,1961,1962,1963,1964,1965,...,2015,2016,2017,2018,2019,2020,2021,2022,2023,Unnamed: 68
0,Aruba,ABW,GDP (current US$),NY.GDP.MKTP.CD,NaN,NaN,NaN,NaN,NaN,NaN,...,2.962907e+09,2.983635e+09,3.092429e+09,3.276184e+09,3.395799e+09,2.558906e+09,3.103184e+09,3.544708e+09,NaN,NaN
1,Africa Eastern and Southern,AFE,GDP (current US$),NY.GDP.MKTP.CD,NaN,NaN,NaN,NaN,NaN,NaN,...,8.992556e+11,8.297383e+11,9.399593e+11,1.012521e+12,1.006191e+12,9.288802e+11,1.086531e+12,1.185138e+12,NaN,NaN
2,Afghanistan,AFG,GDP (current US$),NY.GDP.MKTP.CD,5.377778e+08,5.488889e+08,5.466667e+08,7.511112e+08,8.000000e+08,1.006667e+09,...,1.913422e+10,1.811657e+10,1.875346e+10,1.805322e+10,1.879944e+10,1.995593e+10,1.426650e+10,NaN,NaN,NaN
3,Africa Western and Central,AFW,GDP (current US$),NY.GDP.MKTP.CD,1.041165e+10,1.113592e+10,1.195171e+10,1.268581e+10,1.384900e+10,1.487476e+10,...,7.692632e+11,6.921149e+11,6.856303e+11,7.681582e+11,8.234056e+11,7.869624e+11,8.449275e+11,8.753937e+11,NaN,NaN
4,Angola,AGO,GDP (current US$),NY.GDP.MKTP.CD,NaN,NaN,NaN,NaN,NaN,NaN,...,9.049642e+10,5.276162e+10,7.369016e+10,7.945069e+10,7.089796e+10,4.850156e+10,6.650513e+10,1.067828e+11,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
261,Kosovo,XKX,GDP (current US$),NY.GDP.MKTP.CD,NaN,NaN,NaN,NaN,NaN,NaN,...,6.295848e+09,6.682677e+09,7.180765e+09,7.878760e+09,7.899738e+09,7.717145e+09,9.412034e+09,9.409474e+09,NaN,NaN
262,"Yemen, Rep.",YEM,GDP (current US$),NY.GDP.MKTP.CD,NaN,NaN,NaN,NaN,NaN,NaN,...,4.244449e+10,3.131782e+10,2.684223e+10,2.160616e+10,NaN,NaN,NaN,NaN,NaN,NaN
263,South Africa,ZAF,GDP (current US$),NY.GDP.MKTP.CD,8.748597e+09,9.225996e+09,9.813996e+09,1.085420e+10,1.195600e+10,1.306899e+10,...,3.467098e+11,3.235855e+11,3.814488e+11,4.052607e+11,3.893300e+11,3.382914e+11,4.201178e+11,4.052709e+11,NaN,NaN
264,Zambia,ZMB,GDP (current US$),NY.GDP.MKTP.CD,7.130000e+08,6.962857e+08,6.931429e+08,7.187143e+08,8.394286e+08,1.082857e+09,...,2.125122e+10,2.095841e+10,2.587360e+10,2.631151e+10,2.330867e+10,1.811064e+10,2.209642e+10,2.916378e+10,NaN,NaN


In [26]:
metadata = pd.read_csv('datos/Metadata_Country_API_NY.GDP.MKTP.CD_DS2_en_csv_v2_397725.csv')
metadata

,Country Code,Region,IncomeGroup,SpecialNotes,TableName,Unnamed: 5
0,ABW,Latin America & Caribbean,High income,NaN,Aruba,NaN
1,AFE,NaN,NaN,"26 countries, stretching from the Red Sea in t...",Africa Eastern and Southern,NaN
2,AFG,South Asia,Low income,The reporting period for national accounts dat...,Afghanistan,NaN
3,AFW,NaN,NaN,"22 countries, stretching from the westernmost ...",Africa Western and Central,NaN
4,AGO,Sub-Saharan Africa,Lower middle income,The World Bank systematically assesses the app...,Angola,NaN
...,...,...,...,...,...,...
260,XKX,Europe & Central Asia,Upper middle income,NaN,Kosovo,NaN
261,YEM,Middle East & North Africa,Low income,The World Bank systematically assesses the app...,"Yemen, Rep.",NaN
262,ZAF,Sub-Saharan Africa,Upper middle income,Fiscal year end: March 31; reporting period fo...,South Africa,NaN
263,ZMB,Sub-Saharan Africa,Lower middle income,National accounts data were rebased to reflect...,Zambia,NaN


### Planteamiento del análisis exploratorio

> Cómo ha sido el comportamiento del PIB a nivel mundial por regiones y por nivel de ingreso?

### Preparación de los datos

In [27]:
pib = pib.drop(columns = ['Indicator Name', 'Indicator Code', '2023', 'Unnamed: 68'])
pib

,Country Name,Country Code,1960,1961,1962,1963,1964,1965,1966,1967,...,2013,2014,2015,2016,2017,2018,2019,2020,2021,2022
0,Aruba,ABW,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,2.727850e+09,2.790850e+09,2.962907e+09,2.983635e+09,3.092429e+09,3.276184e+09,3.395799e+09,2.558906e+09,3.103184e+09,3.544708e+09
1,Africa Eastern and Southern,AFE,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,9.633398e+11,9.796905e+11,8.992556e+11,8.297383e+11,9.399593e+11,1.012521e+12,1.006191e+12,9.288802e+11,1.086531e+12,1.185138e+12
2,Afghanistan,AFG,5.377778e+08,5.488889e+08,5.466667e+08,7.511112e+08,8.000000e+08,1.006667e+09,1.400000e+09,1.673333e+09,...,2.014642e+10,2.049713e+10,1.913422e+10,1.811657e+10,1.875346e+10,1.805322e+10,1.879944e+10,1.995593e+10,1.426650e+10,NaN
3,Africa Western and Central,AFW,1.041165e+10,1.113592e+10,1.195171e+10,1.268581e+10,1.384900e+10,1.487476e+10,1.584558e+10,1.442849e+10,...,8.340970e+11,8.945050e+11,7.692632e+11,6.921149e+11,6.856303e+11,7.681582e+11,8.234056e+11,7.869624e+11,8.449275e+11,8.753937e+11
4,Angola,AGO,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,1.323391e+11,1.359668e+11,9.049642e+10,5.276162e+10,7.369016e+10,7.945069e+10,7.089796e+10,4.850156e+10,6.650513e+10,1.067828e+11
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
261,Kosovo,XKX,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,6.735329e+09,7.074395e+09,6.295848e+09,6.682677e+09,7.180765e+09,7.878760e+09,7.899738e+09,7.717145e+09,9.412034e+09,9.409474e+09
262,"Yemen, Rep.",YEM,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,4.041523e+10,4.322859e+10,4.244449e+10,3.131782e+10,2.684223e+10,2.160616e+10,NaN,NaN,NaN,NaN
263,South Africa,ZAF,8.748597e+09,9.225996e+09,9.813996e+09,1.085420e+10,1.195600e+10,1.306899e+10,1.421139e+10,1.582139e+10,...,4.008860e+11,3.811989e+11,3.467098e+11,3.235855e+11,3.814488e+11,4.052607e+11,3.893300e+11,3.382914e+11,4.201178e+11,4.052709e+11
264,Zambia,ZMB,7.130000e+08,6.962857e+08,6.931429e+08,7.187143e+08,8.394286e+08,1.082857e+09,1.264286e+09,1.368000e+09,...,2.803724e+10,2.714102e+10,2.125122e+10,2.095841e+10,2.587360e+10,2.631151e+10,2.330867e+10,1.811064e+10,2.209642e+10,2.916378e+10


In [28]:
metadata = metadata.drop(columns=['SpecialNotes', 'TableName', 'Unnamed: 5'])
metadata

,Country Code,Region,IncomeGroup
0,ABW,Latin America & Caribbean,High income
1,AFE,NaN,NaN
2,AFG,South Asia,Low income
3,AFW,NaN,NaN
4,AGO,Sub-Saharan Africa,Lower middle income
...,...,...,...
260,XKX,Europe & Central Asia,Upper middle income
261,YEM,Middle East & North Africa,Low income
262,ZAF,Sub-Saharan Africa,Upper middle income
263,ZMB,Sub-Saharan Africa,Lower middle income


In [29]:
pib = pib.merge(metadata, how='left', on='Country Code')
pib = pib.drop(columns = ['Country Name', 'Country Code'])
pib

,1960,1961,1962,1963,1964,1965,1966,1967,1968,1969,...,2015,2016,2017,2018,2019,2020,2021,2022,Region,IncomeGroup
0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,2.962907e+09,2.983635e+09,3.092429e+09,3.276184e+09,3.395799e+09,2.558906e+09,3.103184e+09,3.544708e+09,Latin America & Caribbean,High income
1,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,8.992556e+11,8.297383e+11,9.399593e+11,1.012521e+12,1.006191e+12,9.288802e+11,1.086531e+12,1.185138e+12,NaN,NaN
2,5.377778e+08,5.488889e+08,5.466667e+08,7.511112e+08,8.000000e+08,1.006667e+09,1.400000e+09,1.673333e+09,1.373333e+09,1.408889e+09,...,1.913422e+10,1.811657e+10,1.875346e+10,1.805322e+10,1.879944e+10,1.995593e+10,1.426650e+10,NaN,South Asia,Low income
3,1.041165e+10,1.113592e+10,1.195171e+10,1.268581e+10,1.384900e+10,1.487476e+10,1.584558e+10,1.442849e+10,1.488050e+10,1.688228e+10,...,7.692632e+11,6.921149e+11,6.856303e+11,7.681582e+11,8.234056e+11,7.869624e+11,8.449275e+11,8.753937e+11,NaN,NaN
4,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,9.049642e+10,5.276162e+10,7.369016e+10,7.945069e+10,7.089796e+10,4.850156e+10,6.650513e+10,1.067828e+11,Sub-Saharan Africa,Lower middle income
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
261,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,6.295848e+09,6.682677e+09,7.180765e+09,7.878760e+09,7.899738e+09,7.717145e+09,9.412034e+09,9.409474e+09,Europe & Central Asia,Upper middle income
262,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,4.244449e+10,3.131782e+10,2.684223e+10,2.160616e+10,NaN,NaN,NaN,NaN,Middle East & North Africa,Low income
263,8.748597e+09,9.225996e+09,9.813996e+09,1.085420e+10,1.195600e+10,1.306899e+10,1.421139e+10,1.582139e+10,1.712479e+10,1.925699e+10,...,3.467098e+11,3.235855e+11,3.814488e+11,4.052607e+11,3.893300e+11,3.382914e+11,4.201178e+11,4.052709e+11,Sub-Saharan Africa,Upper middle income
264,7.130000e+08,6.962857e+08,6.931429e+08,7.187143e+08,8.394286e+08,1.082857e+09,1.264286e+09,1.368000e+09,1.605857e+09,1.965714e+09,...,2.125122e+10,2.095841e+10,2.587360e+10,2.631151e+10,2.330867e+10,1.811064e+10,2.209642e+10,2.916378e+10,Sub-Saharan Africa,Lower middle income


In [31]:
pib.shape

(266, 65)

En este punto, el DataFrame 'pib' contiene:

* Información de 266 países
* Registro histórico de 63 años
* Nivel de ingreso de cada país

Se require de otros DataFrames:

1. Información a nivel mundial
2. Información a nivel regional
3. Información a nivel de ingreso

### Nivel mundial

In [36]:
df_mundial = pib.iloc[:,:-2].sum(axis=0).to_frame('PIB_MUNDIAL')
df_mundial

,PIB_MUNDIAL
1960,9.159248e+12
1961,9.488805e+12
1962,1.004931e+13
1963,1.088184e+13
1964,1.201437e+13
...,...
2018,7.162767e+14
2019,7.270794e+14
2020,7.047404e+14
2021,8.083171e+14


### Nivel regional

In [37]:
pib['Region'].value_counts()

Region
Europe & Central Asia         58
Sub-Saharan Africa            48
Latin America & Caribbean     42
East Asia & Pacific           37
Middle East & North Africa    21
South Asia                     8
North America                  3
Name: count, dtype: int64

In [41]:
df_region = pib.loc[:, pib.columns != 'IncomeGroup'].groupby('Region').sum()
df_region

,1960,1961,1962,1963,1964,1965,1966,1967,1968,1969,...,2013,2014,2015,2016,2017,2018,2019,2020,2021,2022
Region,,,,,,,,,,,,,,,,,,,,,
East Asia & Pacific,1.478183e+11,1.482015e+11,1.511291e+11,1.685134e+11,1.932490e+11,2.155872e+11,2.409084e+11,2.666266e+11,2.936665e+11,3.381323e+11,...,2.088253e+13,2.153795e+13,2.144455e+13,2.221594e+13,2.372039e+13,2.585912e+13,2.640134e+13,2.646204e+13,3.034158e+13,2.992163e+13
Europe & Central Asia,2.721645e+11,2.959798e+11,3.230898e+11,3.570155e+11,3.948901e+11,4.300457e+11,4.792822e+11,5.161538e+11,5.399254e+11,5.971909e+11,...,2.344200e+13,2.377113e+13,2.047005e+13,2.040842e+13,2.165274e+13,2.318846e+13,2.290791e+13,2.215658e+13,2.531724e+13,2.531201e+13
Latin America & Caribbean,3.928545e+10,4.254242e+10,4.616927e+10,4.943352e+10,5.422954e+10,5.819070e+10,6.267997e+10,6.663365e+10,7.146475e+10,7.884629e+10,...,6.358820e+12,6.480877e+12,5.161103e+12,5.032771e+12,5.585604e+12,5.467798e+12,5.390678e+12,4.579386e+12,5.164311e+12,5.855200e+12
Middle East & North Africa,1.410003e+10,1.480261e+10,1.464959e+10,1.645984e+10,1.817257e+10,2.597507e+10,2.744332e+10,2.931795e+10,3.957213e+10,4.379387e+10,...,3.581785e+12,3.602486e+12,3.179422e+12,3.213373e+12,3.352266e+12,3.518412e+12,3.501031e+12,3.139363e+12,3.723674e+12,4.373106e+12
North America,5.838462e+11,6.043242e+11,6.474216e+11,6.837264e+11,7.352851e+11,7.983295e+11,8.762226e+11,9.275238e+11,1.014480e+12,1.099213e+12,...,1.869625e+13,1.936284e+13,1.976918e+13,2.023001e+13,2.113374e+13,2.226561e+13,2.313212e+13,2.272305e+13,2.532968e+13,2.760873e+13
South Asia,4.751003e+10,5.069384e+10,5.410797e+10,6.086094e+10,6.967715e+10,7.483098e+10,6.292454e+10,6.922780e+10,7.255805e+10,7.976519e+10,...,2.389730e+12,2.614764e+12,2.733400e+12,3.010885e+12,3.433944e+12,3.534083e+12,3.658217e+12,3.489924e+12,4.062850e+12,4.372947e+12
Sub-Saharan Africa,2.386910e+10,2.538488e+10,2.704051e+10,2.906610e+10,3.174050e+10,3.449370e+10,3.707675e+10,3.819043e+10,4.063678e+10,4.585800e+10,...,1.794507e+12,1.871652e+12,1.665136e+12,1.515494e+12,1.619323e+12,1.774715e+12,1.822326e+12,1.708368e+12,1.923676e+12,2.052420e+12


### Nivel de ingreso

In [42]:
pib['IncomeGroup'].value_counts()

IncomeGroup
High income            82
Lower middle income    54
Upper middle income    54
Low income             26
Name: count, dtype: int64

In [45]:
df_ningreso = pib.loc[: , pib.columns != 'Region'].groupby('IncomeGroup').sum()
df_ningreso

,1960,1961,1962,1963,1964,1965,1966,1967,1968,1969,...,2013,2014,2015,2016,2017,2018,2019,2020,2021,2022
IncomeGroup,,,,,,,,,,,,,,,,,,,,,
High income,9.338741e+11,9.887594e+11,1.067269e+12,1.149216e+12,1.254270e+12,1.364809e+12,1.509062e+12,1.618782e+12,1.759261e+12,1.935242e+12,...,4.952682e+13,5.058713e+13,4.772146e+13,4.875136e+13,5.097558e+13,5.422557e+13,5.470026e+13,5.328703e+13,5.945977e+13,6.092213e+13
Low income,5.763031e+09,6.122561e+09,6.585119e+09,7.129398e+09,7.605920e+09,8.264422e+09,8.942431e+09,9.891480e+09,1.002983e+10,1.078977e+10,...,3.744767e+11,4.015508e+11,3.949789e+11,3.610349e+11,3.784801e+11,3.944644e+11,3.971420e+11,3.933065e+11,4.196022e+11,4.510040e+11
Lower middle income,7.641193e+10,8.104839e+10,8.231822e+10,9.150706e+10,1.027715e+11,1.173473e+11,1.078378e+11,1.153159e+11,1.232488e+11,1.367416e+11,...,5.604827e+12,5.896071e+12,5.746409e+12,6.004441e+12,6.457237e+12,6.612134e+12,6.884143e+12,6.618962e+12,7.595603e+12,8.147631e+12
Upper middle income,1.048806e+11,9.793161e+10,9.862156e+10,1.076141e+11,1.244038e+11,1.386042e+11,1.519140e+11,1.604349e+11,1.697295e+11,1.897417e+11,...,2.126849e+13,2.187459e+13,2.056000e+13,2.051005e+13,2.268671e+13,2.437603e+13,2.483209e+13,2.395941e+13,2.838804e+13,2.997527e+13
